src/controllers/controllermanager.cpp

```diff
@@ -174,7 +178,20 @@ void ControllerManager::updateControllerList() {
 
     locker.relock();
     if (newDeviceList != m_controllers) {
+        // Check for keyboard controller
+        KeyboardController* pKeyboardController;
+        foreach (Controller* pController, newDeviceList) {
+            pKeyboardController = qobject_cast<KeyboardController*>(pController);
```

Comment:
I removed `KeyboardEnumerator`. `KeyboardController` is now created and deleted directly in `ControllerManager`. This avoids this nasty loop and casting.